File: Python/binarysearch.py
# Binary Search Algorithm in Python

def binary_search(arr, x):
    # your code goes here
    # return the index of the element if found
    return -1  # if not found

    pass


arr = [2, 3, 4, 10, 40]
x = 10
result = binary_search(arr, x)

if result == -1:
    print("Element is not present in array")
else:
    print("Element is present at index", result)

# Output:
# Element found at index 3

File: Rust/binarysearch.rs
// Binary search algorithm in Rust

fn main() {
    let arr = [2, 3, 4, 10, 40];
    let x = 10;
    if binary_search(arr, x) {
        println!("Element found at index {}", x);
    } else {
        println!("Element not found");
    }
}

fn binary_search(arr: [i32; 5], x: i32) -> bool {
    // your code goes here
}

// Output:
// Element found at index 3
